src/widgets/sparkline.rs

```diff
@@ -158,52 +336,85 @@ impl Sparkline<'_> {
+            // - the style is the style of the item, if one is set
+            //
+            // otherwise:
+            // - the height is the total height of the spark area
+            // - the symbol is the empty bar symbol
+            // - the style is the empty bar style
+            let (mut height, symbol, style) = match item {
+                SparkItem {
+                    bar: Some(value),
+                    style,
+                } => {
+                    let height = if max_height == 0 {
+                        0
+                    } else {
+                        *value * u64::from(spark_area.height) * 8 / max_height
+                    };
+                    (height, None, *style)
                 }
-            })
-            .collect::<Vec<u64>>();
-        for j in (0..spark_area.height).rev() {
-            for (i, d) in data.iter_mut().enumerate() {
-                let symbol = match *d {
-                    0 => self.bar_set.empty,
-                    1 => self.bar_set.one_eighth,
-                    2 => self.bar_set.one_quarter,
-                    3 => self.bar_set.three_eighths,
-                    4 => self.bar_set.half,
-                    5 => self.bar_set.five_eighths,
-                    6 => self.bar_set.three_quarters,
-                    7 => self.bar_set.seven_eighths,
-                    _ => self.bar_set.full,
-                };
-                let x = match self.direction {
-                    RenderDirection::LeftToRight => spark_area.left() + i as u16,
-                    RenderDirection::RightToLeft => spark_area.right() - i as u16 - 1,
-                };
-                buf[(x, spark_area.top() + j)]
-                    .set_symbol(symbol)
-                    .set_style(self.style);
+                _ => (
+                    u64::from(spark_area.height) * 8,
+                    Some(self.empty_bar_symbol.as_str()),
+                    Some(self.empty_bar_style),
+                ),
+            };
 
-                if *d > 8 {
-                    *d -= 8;
+            // render the item from top to bottom
+            //
+            // if the symbol is set it will be used for the entire height of the bar, otherwise the
+            // symbol will be determined by the _remaining_ height.
+            //
+            // if the style is set it will be used for the entire height of the bar, otherwise the
+            // sparkline style will be used.
+            for j in (0..spark_area.height).rev() {
+                let symbol = symbol.unwrap_or_else(|| self.symbol_for_height(height));
+                if height > 8 {
+                    height -= 8;
                 } else {
-                    *d = 0;
+                    height = 0;
                 }
+                buf[(x, spark_area.top() + j)]
+                    .set_symbol(symbol)
+                    .set_style(style.unwrap_or(self.style));
```

Comment:
`.set_style(self.style.patch(style.unwrap_or_default()))` seems to be what we want, it will combine the styles by layering the bar style over the width style.